h2/stream.py

```diff
@@ -701,6 +701,22 @@ def __init__(self,
         # The configuration for this stream.
         self.config = config
 
+    def __repr__(self):
+        stream_state_strs = {
+            StreamState.IDLE: "Idle",
+            StreamState.RESERVED_REMOTE: "Reserved Remote",
+            StreamState.RESERVED_LOCAL: "Reserved Local",
+            StreamState.OPEN: "Open",
+            StreamState.HALF_CLOSED_REMOTE: "Half Closed Remote",
+            StreamState.HALF_CLOSED_LOCAL: "Half Closed Local",
+            StreamState.CLOSED: "Closed"
+        }
```

Comment:
There is an associated unit test for this.

It adds value in the sense that printing log messages is easier.  I was merely trying to represent the stream as it's id and current state.  I thought those would be valuable for logging.

I understand the point you have raised here about new stream states being added.  Are you opposed to it outright or if it was made more robust are you open to the idea?